// ng-http-loader/spinkits.js
"use strict";
/*
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
 * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
 * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
 * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var sk_cube_grid_component_1 = require("./sk-cube-grid/sk-cube-grid.component");
var sk_chasing_dots_component_1 = require("./sk-chasing-dots/sk-chasing-dots.component");
var sk_double_bounce_component_1 = require("./sk-double-bounce/sk-double-bounce.component");
var sk_rotating_plane_component_1 = require("./sk-rotating-plane/sk-rotating-plane.component");
var sk_spinner_pulse_component_1 = require("./sk-spinner-pulse/sk-spinner-pulse.component");
var sk_three_bounce_component_1 = require("./sk-three-bounce/sk-three-bounce.component");
var sk_wandering_cubes_component_1 = require("./sk-wandering-cubes/sk-wandering-cubes.component");
var sk_wave_component_1 = require("./sk-wave/sk-wave.component");
exports.Spinkit = {
    skChasingDots: 'sk-chasing-dots',
    skCubeGrid: 'sk-cube-grid',
    skDoubleBounce: 'sk-double-bounce',
    skRotatingPlane: 'sk-rotationg-plane',
    skSpinnerPulse: 'sk-spinner-pulse',
    skThreeBounce: 'sk-three-bounce',
    skWanderingCubes: 'sk-wandering-cubes',
    skWave: 'sk-wave',
};
exports.SPINKIT_COMPONENTS = [
    sk_cube_grid_component_1.SkCubeGridComponent,
    sk_chasing_dots_component_1.SkChasingDotsComponent,
    sk_double_bounce_component_1.SkDoubleBounceComponent,
    sk_rotating_plane_component_1.SkRotatingPlaneComponent,
    sk_spinner_pulse_component_1.SkSpinnerPulseComponent,
    sk_three_bounce_component_1.SkThreeBounceComponent,
    sk_wandering_cubes_component_1.SkWanderingCubesComponent,
    sk_wave_component_1.SkWaveComponent,
];
